protostar/migrator/cheatcodes/migrator_invoke_cheatcode.py

```diff
@@ -0,0 +1,99 @@
+import asyncio
+from typing import Any, Optional
+
+from starknet_py.net.signer import BaseSigner
+from typing_extensions import Protocol
+
+from protostar.commands.test.test_environment_exceptions import CheatcodeException
+from protostar.starknet.cheatcode import Cheatcode
+from protostar.starknet_gateway import (
+    ContractNotFoundException,
+    GatewayFacade,
+    UnknownFunctionException,
+)
+from protostar.utils.data_transformer import CairoOrPythonData
+
+
+class InvokeCheatcodeProtocol(Protocol):
+    # pylint: disable=too-many-arguments
+    def __call__(
+        self,
+        contract_address: int,
+        function_name: str,
+        inputs: Optional[CairoOrPythonData] = None,
+        max_fee: Optional[int] = None,
+        account_address: Optional[str] = None,
+        auto_estimate_fee: bool = False,
+        signer: Optional[BaseSigner] = None,
+    ) -> Any:
+        ...
+
+
+class MigratorInvokeCheatcode(Cheatcode):
+    def __init__(
+        self,
+        syscall_dependencies: Cheatcode.SyscallDependencies,
+        gateway_facade: GatewayFacade,
+        global_account_address: Optional[str],
+        global_signer: Optional[BaseSigner],
+    ):
+        super().__init__(syscall_dependencies)
+        self._gateway_facade = gateway_facade
+        self._global_signer = global_signer
+        self._global_account_address = global_account_address
+
+    @property
+    def name(self) -> str:
+        return "invoke"
+
+    def build(self) -> InvokeCheatcodeProtocol:
+        return self.invoke
+
+    # pylint: disable=too-many-arguments
+    def invoke(
+        self,
+        contract_address: int,
+        function_name: str,
+        inputs: Optional[CairoOrPythonData] = None,
+        max_fee: Optional[int] = None,
+        account_address: Optional[str] = None,
+        auto_estimate_fee: bool = False,
+        signer: Optional[BaseSigner] = None,
+    ):
+        if not max_fee and not auto_estimate_fee:
```

Comment:
does max_fee = 0 make sense?

Reply:
for invoke - not really, since only devnet allows it, so it might lead to incosistent results across devnet/testnet

Reply:
wdyt about adding an explicit error case that max_fee must be > 0 ?

Reply:
done

Reply:
<img width="769" alt="obraz" src="https://user-images.githubusercontent.com/3450050/186658344-31c60c26-0581-4d41-8283-610b34de0014.png">

Reply:
ok, noticed the new commit.